x-pack/plugins/files/public/components/image/image.tsx

```diff
@@ -28,22 +45,59 @@ export interface Props extends ImgHTMLAttributes<HTMLImageElement> {
  * ```
  */
 export const Image = React.forwardRef<HTMLImageElement, Props>(
-  ({ src, alt, onFirstVisible, ...rest }, ref) => {
+  (
+    { src, alt, onFirstVisible, onLoad, onError, meta, wrapperProps, size = 'original', ...rest },
+    ref
+  ) => {
+    const [isLoaded, setIsLoaded] = useState<boolean>(false);
+    const [blurDelayExpired, setBlurDelayExpired] = useState(false);
     const { isVisible, ref: observerRef } = useViewportObserver({ onFirstVisible });
+
+    useEffect(() => {
+      const id = window.setTimeout(() => setBlurDelayExpired(true), 200);
+      return () => window.clearTimeout(id);
+    }, []);
+
+    const knownSize = size ? sizes[size] : undefined;
+
     return (
-      <img
-        {...rest}
-        ref={(element) => {
-          observerRef(element);
-          if (ref) {
-            if (typeof ref === 'function') ref(element);
-            else (ref as MutableRefObject<HTMLImageElement | null>).current = element;
-          }
-        }}
-        // TODO: We should have a lower resolution alternative to display
-        src={isVisible ? src : undefined}
-        alt={alt}
-      />
+      <div
+        css={[
+          css`
+            position: relative;
+            display: inline-block;
```

Comment:
nit: if it is "inline-block", I would expect the element to be `<span>` for whatever reason (I don't have a good reason, [StackOverflow](https://stackoverflow.com/questions/1611065/span-vs-div-inline-block)).

```suggestion
      <span
        css={[
          css`
            position: relative;
            display: inline-block;
```

Reply:
I think in this case we do want an inline-block, I think it communicates the right thing since we are also giving consumers the possibility to overwrite this style and have it behave as a div.